ReactiveCocoa/Swift/Atomic.swift

```diff
@@ -58,19 +58,20 @@ public final class Atomic<Value> {
 	/// Returns the old value.
 	public func modify(@noescape action: (inout Value) throws -> Void) rethrows -> Value {
 		return try withValue { value in
-			try action(&_value)
-			return value
+			let oldValue = value
+			try action(&value)
+			return oldValue
 		}
 	}
-	
+
 	/// Atomically performs an arbitrary action using the current value of the
 	/// variable.
 	///
 	/// Returns the result of the action.
-	public func withValue<Result>(@noescape action: (Value) throws -> Result) rethrows -> Result {
+	public func withValue<Result>(@noescape action: (inout Value) throws -> Result) rethrows -> Result {
```

Comment:
With this change, actions sent to `withValue` can modify the value. That's a big semantic change, and it's not one that I think makes sense. The point of `withValue` is that it's readonly.